core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -152,11 +152,14 @@ public static DataMapJob getEmbeddedJob() {
   /**
    * Prune the segments from the already pruned blocklets.
    */
-  public static void pruneSegments(List<Segment> segments, List<ExtendedBlocklet> prunedBlocklets) {
+  public static void pruneSegments(List<Segment> segments, List<ExtendedBlocklet> prunedBlocklets,
+      boolean clearExistingShards) {
     Set<Segment> validSegments = new HashSet<>();
     for (ExtendedBlocklet blocklet : prunedBlocklets) {
       // Clear the old pruned index files if any present
-      blocklet.getSegment().getFilteredIndexShardNames().clear();
```

Comment:
@shivamasn  FYI:  wrong query result is occurred for hive partition table+bloom because this line of code too. Remove this line or apply this pr makes that fine.

Reply:
@shivamasn   Does revert related changes in #3177 can solve? 

@kunal642   Does reverting method `pruneSegments` have any problem for Distributed Index Server?